ConManTests.java

```diff
@@ -45,58 +33,44 @@ private Contact will() {
          return new Contact("Will", "Curry", "07555555555", "will@emailsite.com");
     }
 
+    private void buildConMan(String streamInfo) {
+        stream = new ByteArrayInputStream(streamInfo.getBytes());
+        output = new ByteArrayOutputStream();
+        writer = new PrintWriter(output);
+        menu = new ConsoleMenu(new ArrayList<>(), stream, writer);
+        consoleUI = new ConsoleUI(new ByteArrayInputStream("Will\nBilly".getBytes()), writer);
+        conMan = new ConMan(menu, consoleUI, stream);
+    }
+
     @Test
     public void canAddToListOfContacts() {
-        stream = new ByteArrayInputStream("Add\nWill\nCurry\nTele\nEmail".getBytes());
-        writer = new PrintWriter(System.out);
-        consoleUI = new ConsoleUI(stream, writer);
-        menu = new ConsoleMenu<ArrayList<Contact>>(menuItems(), stream, writer);
-        conMan = new ConMan(menu, consoleUI);
-        conMan.userSelectItem();
-        assertThat(conMan.allContacts.get(0).firstName(), is("Will"));
+        buildConMan("Add");
+        conMan.start();
+        assertThat(conMan.allContacts().get(0).firstName(), is("Will"));
     }
 
     @Test
     public void searchingFirstNameReturnsAllContactsWithThatFirstName() {
-        stream = new ByteArrayInputStream("Search\nWill\nWill\nWill\nWill".getBytes());
-        writer = new PrintWriter(System.out);
-        consoleUI = new ConsoleUI(stream, writer);
-        menu = new ConsoleMenu<ArrayList<Contact>>(menuItems(), stream, writer);
-        conMan = new ConMan(menu, consoleUI);
-        Contact contact = will();
-        conMan.allContacts.add(contact);
-        conMan.userSelectItem();
-        System.out.println(conMan.userSelectItem());
-        for (Contact will : conMan.userSelectItem()) {
-            assertThat(will.firstName(), is("Will"));
-        }
+        buildConMan("Search");
+        conMan.allContacts().add(will());
```

Comment:
You wouldn't want to leave your ConMan data e.g. allContacts() open to modification by external objects.
I realize this is for test, but your API will have allContacts() available to anybody.

How could you use ConsoleUI to plug in the Contact data?